### PR TITLE
chore: Detect test CurrentPlatform based on TFM override

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -39,6 +39,8 @@
 	
 	<UNO_UWP_BUILD>true</UNO_UWP_BUILD>
 	<DefineConstants Condition="'$(UNO_UWP_BUILD)'!='true'">$(DefineConstants);HAS_UNO_WINUI</DefineConstants>
+    <DefineConstants Condition="'$(UnoTargetFrameworkOverride)'=='MonoAndroid11.0'">$(DefineConstants);CURRENT_PLATFORM_ANDROID</DefineConstants>
```
Comment (MartinZikmund) [inline src/Directory.Build.props:42]: For future-proofing I would also suggest adding `MonoAndroid12.0`, `net6.0-android` and `net6.0-ios` 👍

**Review by MartinZikmund — APPROVED** [review]

File: src/Directory.Build.props [hunk 2]
```diff
@@ -39,6 +39,8 @@
 	
 	<UNO_UWP_BUILD>true</UNO_UWP_BUILD>
 	<DefineConstants Condition="'$(UNO_UWP_BUILD)'!='true'">$(DefineConstants);HAS_UNO_WINUI</DefineConstants>
+	<DefineConstants Condition="$(UnoTargetFrameworkOverride.ToLower().StartsWith('monoandroid')) OR $(UnoTargetFrameworkOverride.ToLower().EndsWith('-android'))">$(DefineConstants);CURRENT_PLATFORM_ANDROID</DefineConstants>
```
Comment (MartinZikmund) [inline src/Directory.Build.props:42]: Does `ToLowerInvariant` work here? Could _very theoretically_ be problem in Turkey for example

**Review by MartinZikmund — APPROVED** [review]

File: src/Directory.Build.props [hunk 3]
```diff
@@ -39,6 +39,8 @@
 	
 	<UNO_UWP_BUILD>true</UNO_UWP_BUILD>
 	<DefineConstants Condition="'$(UNO_UWP_BUILD)'!='true'">$(DefineConstants);HAS_UNO_WINUI</DefineConstants>
+	<DefineConstants Condition="$(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('monoandroid')) OR $(UnoTargetFrameworkOverride.ToLowerInvariant().EndsWith('-android'))">$(DefineConstants);CURRENT_PLATFORM_ANDROID</DefineConstants>
```
Comment (jeromelaban) [inline src/Directory.Build.props:42]: We cannot use `-android` as `EndsWith` for net6 (this is currently incorrect and will change soon). Let's use StartsWith(`net6.0-android`) for now.

File: src/Directory.Build.props [hunk 4]
```diff
@@ -39,6 +39,8 @@
...
+	<DefineConstants Condition="$(UnoTargetFrameworkOverride.ToLowerInvariant().StartsWith('xamarinios'))  OR $(UnoTargetFrameworkOverride.ToLowerInvariant().EndsWith('-ios'))">$(DefineConstants);CURRENT_PLATFORM_IOS</DefineConstants>
```
Comment (jeromelaban) [inline src/Directory.Build.props:43]: Same here.

File: src/SamplesApp/SamplesApp.UITests/Constants.cs [hunk 5]
```diff
@@ -15,6 +15,13 @@ public partial class Constants
 		public const string iOSDeviceNameOrId = "iPad Pro (12.9-inch) (4th generation)";
 
 		// Default active platform when running under Visual Studio test runner
-		public const Platform CurrentPlatform = Platform.Browser;
+		public const Platform CurrentPlatform =
+#if CURRENT_PLATFORM_ANDROID
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Constants.cs:19]:
```suggestion
#if TARGET_FRAMEWORK_OVERRIDE_ANDROID
```

File: src/SamplesApp/SamplesApp.UITests/Constants.cs [hunk 6]
```diff
@@ -15,6 +15,13 @@ public partial class Constants
...
+			Platform.Android;
+#elif CURRENT_PLATFORM_IOS
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Constants.cs:21]:
```suggestion
#elif TARGET_FRAMEWORK_OVERRIDE_IOS
```

**Review by jeromelaban — COMMENTED** [review]
Nice change!

Let's adjust the documentation as well, as it won't be necessary to change this property with this change.

**Pull request merged**